phpBB/phpbb/template/context.php

```diff
@@ -264,6 +264,89 @@ public function assign_block_vars_array($blockname, array $block_vars_array)
 	}
 
 	/**
+	* Find the index for a specified key in the innermost specified block
+	*
+	* @param	string	$blockname	the blockname, for example 'loop'
+	* @param	mixed	$key		Key to search for
+	*
+	* array: KEY => VALUE [the key/value pair to search for within the loop to determine the correct position]
+	*
+	* int: Position [the position to search for]
+	*
+	* If key is false the position is set to 0
+	* If key is true the position is set to the last entry
+	*
+	* @return mixed false if not found, index position otherwise; be sure to test with ===
+	*/
+	public function find_key_index($blockname, $key)
+	{
+		// For nested block, $blockcount > 0, for top-level block, $blockcount == 0
+		$blocks = explode('.', $blockname);
+		$blockcount = sizeof($blocks) - 1;
+
+		$block = $this->tpldata;
+		for ($i = 0; $i < $blockcount; $i++)
```

Comment:
In its current state, this will not loop for the top-level block. I take it that is expected due to it not needing to traverse into the sub blocks?